beam/core/src/main/java/cz/o2/proxima/beam/core/BeamDataOperator.java

```diff
@@ -471,6 +481,23 @@ public boolean hasDirect() {
       // when limit is applied we must create a new source for each input
       ret = factory.apply(desc);
     }
+    if (!typesRegisteredFor(ret.getPipeline())) {
+      registerTypesFor(ret.getPipeline());
+    }
     return ret;
   }
+
+  private void registerTypesFor(Pipeline pipeline) {
+    CoderRegistry registry = pipeline.getCoderRegistry();
+    registry.registerCoderForClass(GlobalWindow.class, GlobalWindow.Coder.INSTANCE);
+    registry.registerCoderForClass(IntervalWindow.class, IntervalWindowCoder.of());
```

Comment:
do we really need to register coders for beam windows? aren't they already registred?

Reply:
unfortunately, they aren't.